src/course-unit/header-navigations/HeaderNavigations.jsx

```diff
@@ -1,27 +1,42 @@
 import PropTypes from 'prop-types';
 import { useIntl } from '@edx/frontend-platform/i18n';
 import { Button } from '@openedx/paragon';
+import { Edit as EditIcon } from '@openedx/paragon/icons';
 
+import { COURSE_BLOCK_NAMES } from '../../constants';
 import messages from './messages';
 
-const HeaderNavigations = ({ headerNavigationsActions }) => {
+const HeaderNavigations = ({ headerNavigationsActions, unitCategory }) => {
   const intl = useIntl();
-  const { handleViewLive, handlePreview } = headerNavigationsActions;
+  const { handleViewLive, handlePreview, handleEdit } = headerNavigationsActions;
 
   return (
     <nav className="header-navigations ml-auto flex-shrink-0">
-      <Button
-        variant="outline-primary"
-        onClick={handleViewLive}
-      >
-        {intl.formatMessage(messages.viewLiveButton)}
-      </Button>
-      <Button
-        variant="outline-primary"
-        onClick={handlePreview}
-      >
-        {intl.formatMessage(messages.previewButton)}
-      </Button>
+      {unitCategory === COURSE_BLOCK_NAMES.vertical.id && (
+        <>
+          <Button
+            variant="outline-primary"
+            onClick={handleViewLive}
+          >
+            {intl.formatMessage(messages.viewLiveButton)}
+          </Button>
+          <Button
+            variant="outline-primary"
+            onClick={handlePreview}
+          >
+            {intl.formatMessage(messages.previewButton)}
+          </Button>
+        </>
+      )}
+      {unitCategory === COURSE_BLOCK_NAMES.libraryContent.id && (
+        <Button
+          iconBefore={EditIcon}
+          variant="outline-primary"
+          onClick={handleEdit}
+        >
+          {intl.formatMessage(messages.editButton)}
```

Comment:
This edit button is not working for me. Nothing happens when I click it.

![Screenshot 2024-11-27 at 2 51 55 PM](https://github.com/user-attachments/assets/76e15e97-f345-422d-a070-0ec13598665b)

Reply:
The edit functionality will be realized in pull request https://github.com/openedx/frontend-app-authoring/pull/1445 and then adopted to that page in pull request https://github.com/openedx/frontend-app-authoring/pull/1492 (if needed).